app/models/course.rb

```diff
@@ -905,32 +917,8 @@ def name_and_code
     "#{name} (#{course_code})"
   end
 
-  def available_placements
-    @available_placements ||= site_statuses
-                              .joins(:site)
-                              .where(
-                                status: [SiteStatus.statuses[:running], SiteStatus.statuses[:new_status]],
-                                publish: SiteStatus.publishes[:published],
-                                vac_status: [
-                                  SiteStatus.vac_statuses[:both_full_time_and_part_time_vacancies],
-                                  SiteStatus.vac_statuses[:part_time_vacancies],
-                                  SiteStatus.vac_statuses[:full_time_vacancies]
-                                ]
-                              )
-                              .where.not(
-                                "site.address1 IN ('', NULL)
-                                AND site.address2 IN ('', NULL)
-                                AND site.address3 IN ('', NULL)
-                                AND site.town IN ('', NULL)
-                                AND site.address4 IN ('', NULL)
-                                AND site.postcode IN ('', NULL)
-                                AND site.latitude IS NULL
-                                AND site.longitude IS NULL"
-                              )
-  end
-
   def available_placements_count
-    @available_placements_count ||= available_placements.count
+    @available_placements_count ||= site_statuses.size
```

Comment:
I verified and all the site statuses are similar to the query that was doing before. I had to call `.size` because this is preloaded from Active Record. If we use `.count` will be N+1 query on the search results.